### PR TITLE
#215 country view list completed

### LINKED ISSUE #215: [Counties] View counties list
Add a new page in the admin for managing counties and polling stations, it should be displayed in the header and called 'Polling stations'.

Backend part was implemented in https://github.com/code4romania/monitorizare-vot/issues/254 . The endpoint for viewing counties list is `GET api/v1/county`

UI available here: https://www.figma.com/file/61NHR7fwmV1Dsv4Pb6F8T7/code4ro_mv_user-management_2018.04.21-(Copy)?node-id=754%3A0

Linked to UI issue for styleguide https://github.com/code4romania/monitorizare-vot-ong/issues/209 . It would be best if the styleguide issues were implemented first so the elements could be reused here.

- [ ] display the counties list
- [ ] display an empty state if no counties are returned by the API
- [ ] add the possibility of filtering counties by name or county code

Counties data model:
- id, code, name, numberOfPollingStations, Diaspora flag, Order

### PR DESCRIPTION
**Pull request by Botosio**
### #215 
New Polling Station List to View

Closes #215

### How has it been tested?

Tested locally the get and filter options: screen shots below
image

All items in QA
image

TODO:
Asking for new Requirement on table actions.
Buttons on top (import/export/new) are a part of a different requirement.

@aniri

### CODE REVIEW

**Comment by Botosio** [comment]
No polling stations example.
image

**Comment by aniri** [comment]
@Botosio the new page looks great 🚀  

There's a small problem though: the counties dropdown from the answers page is not loading properly:
image

Also, if you have the time, could you please make sure that "county" is used instead of "country". I see we have both terms now 😞

**Comment by Botosio** [comment]
I'll double check that

**Comment by Botosio** [comment]
This should be fixed now.

File: src/app/answers/answers/answers.component.ts [hunk 1]
```diff
@@ -63,9 +63,14 @@ export class AnswersComponent implements OnInit {
     map(threads => threads.map(c => ({ ...c, locationType: (c as any).urbanArea ? 'Urban' : 'Rural' })))
   );
   filters$: Observable<AnswerFilters> = this.store.select(getFilters);
-  counties$: Observable<Partial<County>[]> = this.store.select(getCounties).pipe(
+  /* counties$: Observable<Partial<County>[]> = this.store.select(getCounties).pipe(
```
Comment (idormenco) [inline src/app/answers/answers/answers.component.ts]: remove commented code

File: src/app/answers/answers/answers.component.ts [hunk 2]
```diff
@@ -102,7 +114,7 @@ export class AnswersComponent implements OnInit {
     // TODO: call proper API
```
Comment (idormenco) [inline src/app/answers/answers/answers.component.ts:110]: is this todo still relevant?

File: src/app/components/polling-stations/polling-stations.component.html [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                  <button class="btn bg-light dropdown-toggle-split" ngbDropdownToggle></button>
+                  <div class="dropdown-menu dropdown-menu-right" ngbDropdownMenu>
+                    <button ngbDropdownItem (click)="deleteForm(form)" class="text-danger" style="height: 40px;">
+                      <img src="/assets/forms/icon-delete.png" alt="">
+                      <span style="padding-left: 10px">{{ "DELETE" | translate }}</span>
+                    </button>
+                    <!-- <button *ngIf="form.draft" ngbDropdownItem (click)="setFormDraftStatus(form, false)" class="text"
+                          style="height: 40px;">
+                    <img src="/assets/forms/icon-publish.png" alt="">
+                    <span style="padding-left: 10px">{{ "FORM_PUBLISH" | translate }}</span>
```
Comment (idormenco) [inline src/app/components/polling-stations/polling-stations.component.html]: remover commented code

File: src/app/components/polling-stations/polling-stations.component.html [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  </div>
+
+</div>
+
+
+
+
+
+<!--
+<div class="row">
```
Comment (idormenco) [inline src/app/components/polling-stations/polling-stations.component.html]: remove

File: src/app/store/county/county.effects.ts [hunk 5]
```diff
@@ -13,26 +13,39 @@ import { getCounties } from './county.selectors';
...
   );
 
-  constructor (
-    private actions$: Actions,
-    private apiService: ApiService,
-    private store: Store<AppState>
-  ) { }
+  @Effect() getPollingStations$ = this.actions$.pipe(
+    ofType(CountryActionTypes.FETCH_COUNTRIES_FOR_POLLING_STATIONS),
+    // withLatestFrom(this.store.select(getCounties)),
```
Comment (idormenco) [inline src/app/store/county/county.effects.ts]: remove

**Review by idormenco — CHANGES_REQUESTED** [review]

File: src/app/answers/answers/answers.component.ts (hunk 2 above)
Comment (Botosio) [reply src/app/answers/answers/answers.component.ts:110]: Not sure, doesn't look like the sort works.

image

File: src/app/store/county/county.effects.ts (hunk 5 above)
Comment (Botosio) [reply src/app/store/county/county.effects.ts]: Done

File: src/app/components/polling-stations/polling-stations.component.html (hunk 4 above)
Comment (Botosio) [reply src/app/components/polling-stations/polling-stations.component.html]: Done

File: src/app/components/polling-stations/polling-stations.component.html (hunk 3 above)
Comment (Botosio) [reply src/app/components/polling-stations/polling-stations.component.html]: Done

File: src/app/answers/answers/answers.component.ts (hunk 1 above)
Comment (Botosio) [reply src/app/answers/answers/answers.component.ts]: Done

Comment (Botosio) [reply src/app/answers/answers/answers.component.ts]: Done

**Review by idormenco — APPROVED** [review]
lgtm

**Pull request merged**